config/post-install/storage-version-migrator.yaml

```diff
@@ -37,6 +37,4 @@ spec:
         - name: migrate
           image: ko://knative.dev/eventing/vendor/knative.dev/pkg/apiextensions/storageversion/cmd/migrate
           args:
-            - "sinkbindings.sources.knative.dev"
-            - "apiserversources.sources.knative.dev"
-            - "containersources.sources.knative.dev"
+            - "pingsources.sources.knative.dev"
```

Comment:
I know this is confusing, but likely the whole /post-install needs to be recreated. @lionelvillard

Reply:
@zhongduo why?

/lgtm

Reply:
I'm not sure I follow your question. this is equivalent to deleting existing post-install folder and create new one. 

Also I have one question regarding the new folder structure:

1. User upgrade from 0.19 -> 0.20, the post-install job will be run:

```
k get job -n knative-eventing
NAME                        COMPLETIONS   DURATION   AGE
storage-version-migration   1/1           3s         6m52s
```

2. If the job still exists and user upgrade from 0.20 -> 0.21, since the job with the same name is already done, IIUC it won't run again. should I choose a different Job name?

Reply:
@eclipselu I realized it is equivalent too. My original concern is that I didn't see a remove of the directory and not sure if there is other logic in the directory. e.g. maybe tomorrow when they are cleaning up, they will remove the directory without checking. But as long as this file is the only job now, I agree we can reuse it.

Reply:
@eclipselu this is what `ttlSecondsAfterFinished` does and we already use it:
https://github.com/knative/eventing/blob/0f2f5671d73814f327353c019c5758bd87d1db6d/config/post-install/storage-version-migrator.yaml#L24 
However, it's an alpha feature which means it is not available in all K8s clusters, so I think we should use [generated values](https://kubernetes.io/docs/reference/using-api/api-concepts/#generated-values).

Reply:
> @eclipselu I realized it is equivalent too. My original concern is that I didn't see a remove of the directory and not sure if there is other logic in the directory. e.g. maybe tomorrow when they are cleaning up, they will remove the directory without checking. But as long as this file is the only job now, I agree we can reuse it.

thanks, I see what you mean now. I'm leaning towards reusing the job but open to both options.

Reply:
> @eclipselu this is what `ttlSecondsAfterFinished` does and we already use it:
> https://github.com/knative/eventing/blob/0f2f5671d73814f327353c019c5758bd87d1db6d/config/post-install/storage-version-migrator.yaml#L24
> 
> 
> However, it's an alpha feature which means it is not available in all K8s clusters, so I think we should use [generated values](https://kubernetes.io/docs/reference/using-api/api-concepts/#generated-values) and `ttlSecondsAfterFinished`.

Oh, sorry I did not realize that we already have this. It seems `ttlSecondsAfterFinished` is not enabled on my cluster, will investigate on generated values, thanks for the pointer!

Reply:
It seems we cannot use generateName with `kubectl apply -f`, https://github.com/kubernetes/kubernetes/issues/44501
I also tried my self, and it did not work: 

```
ko apply -f $GOPATH/src/knative.dev/eventing/config/post-install
2021/01/15 13:43:14 Using base gcr.io/distroless/static:nonroot for knative.dev/eventing/vendor/knative.dev/pkg/apiextensions/storageversion/cmd/migrate
2021/01/15 13:43:15 Building knative.dev/eventing/vendor/knative.dev/pkg/apiextensions/storageversion/cmd/migrate for linux/amd64
2021/01/15 13:43:20 Publishing us.gcr.io/danglu-knative-dev-290519/migrate-86cf205285e60d2a8b66a01f2a57a58b:latest
2021/01/15 13:43:20 existing blob: sha256:72164b581b02b1eb297b403bcc8fc1bfa245cb52e103a3a525a0835a58ff58e2
2021/01/15 13:43:20 existing blob: sha256:5749e56bea7178768b00aac0bf087558fccd5b8e0c807610618be7568459f359
2021/01/15 13:43:20 existing blob: sha256:bb06e95bf79a0e27cf4855dff6e1091225dfd7bfd3e3f24c781139388f4a7fc4
2021/01/15 13:43:20 existing blob: sha256:5338a849ff04b47d70707f299ce9baf5746eb484e14abb8714d69737f899517e
2021/01/15 13:43:21 us.gcr.io/danglu-knative-dev-290519/migrate-86cf205285e60d2a8b66a01f2a57a58b:latest: digest: sha256:150e69e03ac3457dd468348acdc57790e68639c0e8fadcbdb648737fbe803232 size: 751
2021/01/15 13:43:21 Published us.gcr.io/danglu-knative-dev-290519/migrate-86cf205285e60d2a8b66a01f2a57a58b@sha256:150e69e03ac3457dd468348acdc57790e68639c0e8fadcbdb648737fbe803232
clusterrole.rbac.authorization.k8s.io/knative-eventing-post-install-job-role unchanged
serviceaccount/knative-eventing-post-install-job unchanged
clusterrolebinding.rbac.authorization.k8s.io/knative-eventing-post-install-job-role-binding unchanged
error: from storage-version-migration-: cannot use generate name with apply
```

Reply:
Is this resolved now? @pierDipi

Reply:
> It seems we cannot use generateName with kubectl apply -f

Ok, then we should manually change the name until `ttlSecondsAfterFinished` becomes a beta feature.

Reply:
Added a prefix to the Job name.